src/components/USMap/USMap.tsx

```diff
@@ -26,70 +33,86 @@ const stateFipsCodes = Object.keys(statesByFips);
  * This is special cased from the normal map display. The mariana islands are
  * small enough that simply showing the islands is not a UX that works.
  */
-const MarianaIslands = ({ fill }: { fill: string }) => (
+const MarianaIslands = (props: React.SVGProps<SVGRectElement>) => (
   <g transform="translate(40, 395) scale(0.8)" tabIndex={-1}>
-    <rect width="20" height="20" fill={fill} />
+    <rect width="20" height="20" {...props} />
     <text transform="translate(25, 15)">CNMI</text>
   </g>
 );
 
 interface USMapProps {
   showCounties: boolean;
   getFillColor: (fipsCode: string) => string;
-  renderTooltip: (stateFipsCode: string) => ReactNode;
+  renderTooltip: (
+    stateFipsCode: string,
+    tooltipMode: TooltipMode,
+  ) => ReactElement | string;
+  tooltipMode: TooltipMode;
+  activeRegionStyle: ActiveRegionStyle;
 }
 
 const projection = geoAlbersUsaTerritories().scale(1070).translate([400, 300]);
 
 const USMap = React.memo(
-  ({ showCounties, getFillColor, renderTooltip }: USMapProps) => {
-    const [tooltipContent, setTooltipContent] = useState<ReactNode>(null);
-
-    const onMouseLeave = () => setTooltipContent('');
-
-    const mapContent = useMemo(
-      () => (
-        <USStateMapWrapper $showCounties={showCounties}>
-          <ComposableMap data-tip="" projection={projection} height={500}>
-            <g transform="translate(0, -50)">
-              {showCounties && <MapCounties getFillColor={getFillColor} />}
-              <Geographies geography={stateGeographies}>
-                {({ geographies }) =>
-                  geographies
-                    .filter(geo => stateFipsCodes.includes(geo.id))
-                    .map(geo => {
-                      const fipsCode = geo.id;
-                      // we can coerce this to a state safely because we're only
-                      // dealing with FIPS codes that are from states, because of that filter.
-                      const state = findStateByFipsCodeStrict(
-                        fipsCode,
-                      ) as StateType;
+  ({
+    showCounties,
+    getFillColor,
+    renderTooltip,
+    tooltipMode,
+    activeRegionStyle,
+  }: USMapProps) => {
+    const [activeState, setActiveState] = useState<State | null>(null);
+    const regionGeographies = useMemo(() => {
```

Comment:
I refactored the memoized `mapContent` (basically the entire map) into just `regionGeographies` since we need to add a geometry for the active region when using `ActiveRegionStyle.OUTLINE`...  This allows us to still memoize (cache) the bulk of the map even though the outline region changes on hover change.